lib/Setup Thermostats.js

```diff
@@ -58,11 +58,12 @@ exports.setupThermostat = function (that, services)
 				CurrentState.updateValue(currentOperatingState);
 			});	
 
-		var TargetState 		= ThermostatService.getCharacteristic(Characteristic.TargetHeatingCoolingState)
+		var TargetState = ThermostatService.getCharacteristic(Characteristic.TargetHeatingCoolingState)
 			.updateOnHubEvents(that.currentAccessory.id, "thermostatMode")
 			.setInitialValue(TargetStateReportMap.get(that.currentAccessory.attributes.thermostatMode))
 			.on('set', function(newHomekitValue, callback, context)
 			{
+				this.targetState = newHomekitValue; // waiting for callback isn't returned before setTargetTemp in scenes

```

Comment:
Not sure what this addition at line 66 is meant to do - I think the "this" you are using is local to the function "function(newHomekitValue, callback, context)" being called in the .on('set ... on line 64 so the value of this.targetState can't be used outside of the code on lines 66-68?

Reply:
I’m definitely seeing issues still where the plug-in is requesting setCoolingThreshold even after it’s received a heat mode request.